cms/src/services/gdc-importer/VariantImporter.js

```diff
@@ -92,119 +143,426 @@ const Import = function(modelName, fileId, filename) {
+    queue = [
+      ...queue,
+      ...modelsStatus[GDC_MODEL_STATES.singleNgcm].map(modelName => {
+        const fileData = getCancerModelFilesFromMafFileData(modelsFileData[modelName], true)[0];
+        return Import({
+          modelName,
+          fileId: fileData.fileId,
+          filename: fileData.filename,
+        });
+      }),
+      ...modelsStatus[GDC_MODEL_STATES.singleNgcmPlusEngcm].map(modelName => {
+        const fileData = getCancerModelFilesFromMafFileData(modelsFileData[modelName], true)[0];
+        return Import({
+          modelName,
+          fileId: fileData.fileId,
+          filename: fileData.filename,
+        });
+      }),
+    ];
+
+    // Start queue
+    if (!running) {
+      start();
+    }
+
+    return { success: true, startTime: Date.now() };
+  };
+
   const stopImport = async modelName => {
     // In case we get into an invalid state with multiple imports for a given model name,
     //   we'll use filter to get the whole list of them.
-    const targets = imports.filter(i => i.modelName === modelName);
+    const targets = queue.filter(i => i.modelName === modelName);
     if (targets.length) {
       targets.forEach(target => target.stop());
+      stopped = [...stopped, ...targets];
+      queue = queue.filter(i => i.modelName !== modelName);
     }
 
-    cleanImportsList();
+    cleanLists();
 
-    return targets.map(summarizeImport);
+    return targets.map(target => target.getData());
   };
 
-  const getImports = () => {
-    cleanImportsList();
+  const stopBulkImport = async () => {
+    pause();
+
+    const targets = queue;
+    if (targets.length) {
+      targets.forEach(target => target.stop());
+      stopped = [...stopped, ...targets];
+      emptyQueue();
+    }
 
-    return imports.map(summarizeImport);
+    cleanLists();
+
+    return targets.map(target => target.getData());
   };
 
-  const acknowledgeCompleted = modelName => {
-    const targets = imports.filter(
-      i =>
-        i.modelName === modelName &&
-        [ImportStatus.complete, ImportStatus.error].includes(i.getData().status),
-    );
+  const getStatus = () => {
+    cleanLists();
+    return {
+      queue: queue.map(i => i.getData()),
+      failed: failed.map(i => i.getData()),
+      stopped: stopped.map(i => i.getData()),
+      success: success.map(i => i.getData()),
+      running,
+    };
+  };
+
+  const acknowledge = modelName => {
+    const targets = [
+      ...failed.filter(i => i.modelName === modelName),
+      ...stopped.filter(i => i.modelName === modelName),
+      ...success.filter(i => i.modelName === modelName),
+    ];
     if (targets.length) {
       targets.forEach(target => target.acknowledge());
     }
 
-    cleanImportsList();
+    cleanLists();
+
+    return targets.map(target => target.getData());
+  };
+
+  const acknowledgeBulk = modelNames => {
+    let targets = [];
+
+    modelNames.forEach(modelName => {
+      targets = [
```

Comment:
After this forEach loop, `targets` will only have the items for the last model. I think this should be `targets.concat([...lists])`

Reply:
Thanks for pointing this out! I tested it and it seems to be working fine since I'm spreading the current `targets` list in first, I'm getting back all the targets in the response